plugin/evm/validators_cache_test.go

```diff
@@ -0,0 +1,40 @@
+package evm
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWithCacheSize(t *testing.T) {
+	t.Run("Signature", func(t *testing.T) {
+		var (
+			cacheOption       interface{}
+			signatureExpected = "evm.CacheOption"
+		)
+
+		for i := 0; i < 10; i++ {
+			slots := rand.Intn(1000)
+			cacheOption = WithCacheSize(uint(slots))
+			signature := reflect.TypeOf(cacheOption).String()
+
+			assert.Equal(t, signature, signatureExpected)
+		}
+	})
+
+	t.Run("CfgMutation", func(t *testing.T) {
+		testCfg := CacheConfig{
+			CacheSize: uint(9),
+		}
+
+		for i := 0; i < 25; i++ {
+			slots := rand.Intn(1000)
+			cacheOption := WithCacheSize(uint(slots))
+			cacheOption(&testCfg)
+
+			assert.Equal(t, testCfg.CacheSize, uint(slots))
+		}
+	})
+}
```

Comment:
I'm not sure why your test checks the returned function's type 🤔 Why would it change upon calling the function?

Testing this option func should be as simple as the following IMO:

```suggestion
func TestWithCacheSize(t *testing.T) {
	wantCacheSize := 42

	cfg := CacheConfig{}
	WithCacheSize(wantCacheSize)(&cfg)

	assert.Equal(t, wantCacheSize, cfg.CacheSize)
}
```

Reply:
Why you removed the tested behavior ?

Reply:
Yes, corrected https://github.com/flare-foundation/coreth/pull/21/commits/4b455472b4f4659e87bf3e6827c1bd26078e71ee#diff-53d507a9b6fbf6d169f0b9686ad727450a293914f998f4a17c524fa84623c4d7L19
Thanks!